pkg/segment/structs/segstructs.go

```diff
@@ -835,6 +837,44 @@ func (qa *QueryAggregators) HasRexBlockInChainWithStats() bool {
 	return false
 }
 
+// To determine whether to fetch all the columns by time range.
+// Currently, it is only used in the case of FillNullExpr
+func (qa *QueryAggregators) AllColumnsByTimeRangeIsRequired() bool {
+	return qa != nil && qa.HasFillNullExprInChain()
+}
+
+func (qa *QueryAggregators) HasFillNullExpr() bool {
+	if qa != nil && qa.OutputTransforms != nil && qa.OutputTransforms.LetColumns != nil && qa.OutputTransforms.LetColumns.FillNullRequest != nil {
+		return true
+	}
+	return false
+}
+
+func (qa *QueryAggregators) HasFillNullExprInChain() bool {
+	if qa == nil {
```

Comment:
You don't have to change anything here for this PR. But we have a lot of these `Has_blank_InChain()` functions with the same exact structure. At some point we could make a higher-level of this and then for the implementation of this we could do just
```
func (qa *QueryAggregators) HasFillNullExprInChain() bool {
    return qa.hasInChain(qa.HasFillNullExpr)
}

Reply:
Probably we should create this as a internal JIRA issue or can we create this as a Github issue?

Reply:
I can make a GitHub Issue for it